test/connection_factory/database_replica_url_test.rb

```diff
@@ -0,0 +1,48 @@
+require "test_helper"
+
+class DatabaseReplicaUrlTest < Minitest::Test
+
+  test_config = { replica: { host: 'test_host', database: 'test_db' } }
+
+  test "use standard config/database.yml spec when DATABASE_REPLICA_URL is not defined" do
+
+    ENV.delete('DATABASE_REPLICA_URL')
```

Comment:
This line is the cause of flaky test.
If this line executes before `old_config_valid_test.rb`, the test successes always.